cmd/backup/hooks.go

```diff
@@ -6,6 +6,8 @@ package main
 import (
 	"fmt"
 	"sort"
+
+	u "github.com/offen/docker-volume-backup/cmd/backup/utilities"
```

Comment:
I'm not sure yet what to think about splitting this into multiple packages.

Go has very strong opinions on how to do this and there are also some conventions around "importability" that we should be following if we wanted to do this.

## Package location

Right now, the newly introduced packages are in `cmd` which looks as if they contain a runnable program when they don't. The proper location for such a "utilities" package would be `github.com/offen/docker-volume-backup/utilities` (or even `github.com/offen/utilities`). Doing this would create a new problem though: the package is now importable by any other Go project, which I don't really want to happen (API stability and such). The Go way of doing this would be making these packages "internal" which prevents foreign imports from happening. The proper location for this package would therefore probably be `github.com/offen/docker-volume-backup/internal/utilities`.

See this article: https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface.

## Package contents

I'm not sure if the split into packages like `types` and `util` makes much sense here, as there is only a single consumer, and it makes the code more verbose. I could see the need for a `storage` package (which the also contains the types), but I'm not so sure about `types` and `util`. "If in doubt, keep it dumb".

Some extra context:
- https://github.com/golang/go/wiki/CodeReviewComments#package-names
- https://go.dev/blog/package-names

---

I'm sure you spent a lot of time and thought on this. What's your opinion here?

Reply:
Thank you for the detailed information.
I agree, that it makes more sense to move things like utilities in an internal package. I somehow didn't really think about that.

The only difference between the contents of `types` and `util` are, that `types` contains only structs without any logic and `util` the contrary. That was my thinking there, but it wouldn't really hurt to keep them together either.

Reply:
Another option would be not having these packages altogether. As there is only a single consumer right now, having this in different packages just creates unnecessary questions when adding new code like "where should this piece of code go" when the benefit is not really all that great.

Reply:
The initial reason for me to have them both in separate packages was, that both `main` and `storages` depend on it.
If we would move `util` and `types` back into `main`, we would have cycling dependency between ´main` and `storages`.

Reply:
Ah ok, I didn't see that. So how about we proceed like this: after changing things as discussed, we see if we can live without an "util" type package (which would be a good sign from a design perspective I guess as it tells us things are nicely separated) and have everything in package main. If that doesn't work out, we add a single internal util package containing things that would create circular imports otherwise. Does this make sense to you too?

Reply:
Sounds good to me!